### PR TITLE
fix: Resolve problem with fallback to location services

### PR DESCRIPTION
**Pull request by dpgraham**

### CODE REVIEW

File: app/src/main/java/io/appium/settings/LocationTracker.java [hunk 1]
```diff
@@ -244,20 +249,24 @@ public synchronized Location getLocation(Context context) {
...
             try {
-                //noinspection deprecation
                 mLocation = LocationServices.FusedLocationApi.getLastLocation(mGoogleApiClient);
                 if (mLocation != null) {
                     // Make sure the fallback is removed after play services connection succeeds
                     if (isLocationManagerConnected()) {
                         stopLocationUpdatesWithoutPlayServices();
                     }
                     return mLocation;
+                } else {
```
Comment (mykola-mokhnach) [inline app/src/main/java/io/appium/settings/LocationTracker.java]: I'd rather flip the if condition

File: app/src/main/java/io/appium/settings/LocationTracker.java [hunk 2]
```diff
@@ -244,20 +249,24 @@ public synchronized Location getLocation(Context context) {
...
+                    stopLocationUpdatesWithPlayServices();
```
Comment (mykola-mokhnach) [inline app/src/main/java/io/appium/settings/LocationTracker.java]: should we always stop google services if the returned location is null? I mean it still might need some time to receive the actual location after being connected

File: app/src/main/java/io/appium/settings/LocationTracker.java (hunk 1 above)
Comment (dpgraham) [reply app/src/main/java/io/appium/settings/LocationTracker.java]: Yeah good thinking.

File: app/src/main/java/io/appium/settings/LocationTracker.java (hunk 2 above)
Comment (dpgraham) [reply app/src/main/java/io/appium/settings/LocationTracker.java]: I was thinking about that. I guess there's no harm having both services running.

**Comment by dpgraham** [comment]
Fixes pushed.

**Pull request merged**